utils/kubeutil.py

```diff
@@ -93,5 +106,101 @@ def extract_kube_labels(self, pods_list, excluded_keys=None):
 
         return kube_labels
 
+    def extract_meta(self, pods_list, field_name):
+        """
+        Exctract fields like `uid` or `name` from the `metadata` section of a
+        list of pods coming from the kubelet API.
+        """
+        uids = []
+        pods = pods_list.get("items") or []
+        for p in pods:
+            value = p.get('metadata', {}).get(field_name)
+            if value is not None:
+                uids.append(value)
+        return uids
+
     def retrieve_pods_list(self):
+        """
+        Retrieve the list of pods for this cluster querying the kubelet API.
+
+        TODO: the list of pods could be cached with some policy to be decided.
+        """
         return retrieve_json(self.pods_list_url)
+
+    def retrieve_metrics(self):
+        """
+        Retrieve metrics from Cadvisor.
+        """
+        return retrieve_json(self.metrics_url)
+
+    def filter_pods_list(self, pods_list, host_ip):
```

Comment:
I suspect this is a remain of the other approach we tried. It could be useful for later in this case. If you think we should keep them around, feel free to.